lms/lmsdb/models.py

```diff
@@ -122,6 +123,62 @@ def on_save_handler(model_class, instance, created):
         instance.password = generate_password_hash(instance.password)
 
 
+class Notification(BaseModel):
+    ID_FIELD_NAME = 'id'
+    MAX_PER_USER = 10
+
+    user = ForeignKeyField(User)
+    created = DateTimeField(default=datetime.now)
+    notification_type = CharField()
+    message_parameters = CharField()
+    related_object_id = IntegerField()
+    marked_read = BooleanField(default=False)
+
+    @property
+    def message_parameters_dict(self) -> dict:
+        return json.loads(self.message_parameters)
+
+    def mark_as_read(self):
+        self.marked_read = True
+        self.save()
+
+    @classmethod
+    def notifications_for_user(
+            cls,
+            for_user: User,
+    ) -> Iterable['Notification']:
+        return cls.select().join(User).filter(
```

Comment:
Why not?

Reply:
For the sake of consistency, because `where` is much more documented and because this is a thing that represent a query (in which you use WHERE clause)